airflow-core/src/airflow/utils/sqlalchemy.py

```diff
@@ -390,7 +392,9 @@ def lock_rows(query: Query, session: Session) -> Generator[None, None, None]:
 
     :meta private:
     """
-    locked_rows = with_row_locks(query, session)
+    select_stmt = cast("Select", query.statement)
+    locked_statement = with_row_locks(select_stmt, session)
+    locked_rows = session.execute(locked_statement).all()
```

Comment:
Do we have to call `all()` here? It’s quite inefficient especially since we don’t actually use the value.